src/View/Form/DocumentContext.php

```diff
@@ -343,11 +344,16 @@ public function error($field)
     {
         $parts = explode('.', $field);
         $entity = $this->entity($parts);
+        $errors = [];
 
         if ($entity instanceof Document) {
-            return $entity->getError(array_pop($parts));
+            $errors = $entity->getError(array_pop($parts));
+
+            if (!$errors && $this->_context['entity'] instanceof Document) {
+                $errors = array_values((array)Hash::get($this->_context['entity']->getErrors(), $field, []));
```

Comment:
Would using `Hash::extract()` let you skip the array cast and `array_values` call?

Reply:
I believe it does yes, took the current line from cake's core [FormContext::error](https://github.com/cakephp/cakephp/blob/master/src/View/Form/FormContext.php#L179) method.